src/DB_Store.php

```diff
@@ -106,6 +184,28 @@ public function fetch_action( $action_id ) {
 		return $this->make_action_from_db_record( $data );
 	}
 
+	/**
+	 * Get raw action data from the database.
+	 *
+	 * @author Jeremy Pry
+	 *
+	 * @param int $action_id
+	 *
+	 * @return \stdClass
+	 */
+	public function get_raw_action( $action_id ) {
```

Comment:
This method forces the calling code to have knowledge of how the action is stored. Meaning the same code can not be used for both old and new data stores (and those yet to come).

Instead of this data store providing raw action data, it can provide the `ActionScheduler_Store::get_last_attempt()` method methods in my other comment for accessing the last attempt date regardless of the underlying storage schema (that will make the unit tests easier too).

The need for that is partly a consequence of the underlying architectural issues with an instance of an action having very little data about itself. But for now, as I think Prospress/action-scheduler#110 showed, we're better off implementing this logic at the data store level for now.